File: Editorial/160_Intersection_of_Two_Linked_Lists/CSharp/2_Hash_Table.cs
public class Solution {
    public ListNode GetIntersectionNode(ListNode headA, ListNode headB) {
        HashSet<ListNode> nodesInB = new HashSet<ListNode>();

        while (headB != null) {
            nodesInB.Add(headB);
            headB = headB.next;
        }

        while (headA != null) {
            // if we find the node pointed to by headA,
            // in our set containing nodes of B, then return the node
            if (nodesInB.Contains(headA)) {
                return headA;
            }

            headA = headA.next;
        }

        return null;
    }
}

File: Editorial/160_Intersection_of_Two_Linked_Lists/Go/2_Hash_Table.go
func getIntersectionNode(headA, headB *ListNode) *ListNode {
    nodesInB := make(map[*ListNode]bool)

    for headB != nil {
        nodesInB[headB] = true
        headB = headB.Next
    }

    for headA != nil {
        // if we find the node pointed to by headA,
        // in our map containing nodes of B, then return the node
        if _, exist := nodesInB[headA]; exist {
            return headA
        }
        headA = headA.Next
    }

    return nil
}
